contract/src/main/java/com/ritense/valtimo/contract/utils/RequestHelper.java

```diff
@@ -34,4 +37,23 @@ public static String getOrigin() {
         return ipList.toString();
     }
 
+    public static ZoneOffset getZoneOffset() {
+        ZoneOffset zoneOffset;
+
+        try {
+            RequestAttributes attribs = RequestContextHolder.getRequestAttributes();
+            HttpServletRequest request = ((ServletRequestAttributes) attribs).getRequest();
+            String zoneOffsetHeader = request.getHeader("Zone-Offset");
+
+            if (zoneOffsetHeader != null) {
+                zoneOffset = ZoneOffset.of(zoneOffsetHeader);
+            } else {
+                throw new Exception();
```

Comment:
it's usually bad practice to use exceptions are part of the normal program flow.You want to use them to signal an exceptional situation to other components that call yours. I would just set the timezone directly here.